app/livechat/server/api/v1/room.js

```diff
@@ -179,3 +181,48 @@ API.v1.addRoute('livechat/room.forward', { authRequired: true }, {
 		API.v1.success(Meteor.runAsUser(this.userId, () => Meteor.call('livechat:transfer', this.bodyParams)));
 	},
 });
+
+API.v1.addRoute('livechat/room.visitor', { authRequired: true }, {
+	put() {
+		try {
+			check(this.bodyParams, {
+				rid: String,
+				visitorId: String,
+			});
+
```

Comment:
Please, check also the following method:
```suggestion
		const user = Users.findOneById(this.userId);
		if (!user) {
			throw new Error('error-user-not-found');
		}

		if (!canAccessRoom(room, user)) {
			throw new Error('error-not-allowed');
		}
```